chain/types/ethtypes/eth_legacy_transaction.go

```diff
@@ -0,0 +1,272 @@
+func (tx *LegacyEthTx) ToUnsignedMessage(from address.Address) (*types.Message, error) {
+	var err error
+	var params []byte
+	if len(tx.Input) > 0 {
+		buf := new(bytes.Buffer)
+		if err = cbg.WriteByteArray(buf, tx.Input); err != nil {
+			return nil, xerrors.Errorf("failed to write input args: %w", err)
+		}
+		params = buf.Bytes()
+	}
+
+	var to address.Address
+	var method abi.MethodNum
+	// nil indicates the EAM, only CreateExternal is allowed
+	if tx.To == nil {
+		method = builtintypes.MethodsEAM.CreateExternal
+		to = builtintypes.EthereumAddressManagerActorAddr
+	} else {
+		method = builtintypes.MethodsEVM.InvokeContract
+		to, err = tx.To.ToFilecoinAddress()
+		if err != nil {
+			return nil, xerrors.Errorf("failed to convert To into filecoin addr: %w", err)
+		}
+	}
+
+	return &types.Message{
+		Version:    0,
+		To:         to,
+		From:       from,
+		Nonce:      uint64(tx.Nonce),
+		Value:      tx.Value,
+		GasLimit:   int64(tx.GasLimit),
+		GasFeeCap:  tx.GasPrice,
+		GasPremium: tx.GasPrice,
+		Method:     method,
+		Params:     params,
+	}, nil
+}
+
+func (tx *LegacyEthTx) ToSignedMessage() (*types.SignedMessage, error) {
+	from, err := tx.Sender()
+	if err != nil {
+		return nil, xerrors.Errorf("failed to calculate sender: %w", err)
+	}
+
+	unsignedMsg, err := tx.ToUnsignedMessage(from)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to convert to unsigned msg: %w", err)
+	}
+
+	siggy, err := tx.Signature()
+	if err != nil {
+		return nil, xerrors.Errorf("failed to calculate signature: %w", err)
+	}
+
+	return &types.SignedMessage{
+		Message:   *unsignedMsg,
+		Signature: *siggy,
+	}, nil
+}
+
+func (tx *LegacyEthTx) packTxFields() ([]interface{}, error) {
+	nonce, err := formatInt(tx.Nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	// format gas price
+	gasPrice, err := formatBigInt(tx.GasPrice)
+	if err != nil {
+		return nil, err
+	}
+
+	gasLimit, err := formatInt(tx.GasLimit)
+	if err != nil {
+		return nil, err
+	}
+
+	value, err := formatBigInt(tx.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	res := []interface{}{
+		nonce,
+		gasPrice,
+		gasLimit,
+		formatEthAddr(tx.To),
+		value,
+		tx.Input,
+	}
+	return res, nil
+}
+
+func (tx *LegacyEthTx) ToRlpUnsignedMsg() ([]byte, error) {
+	packedFields, err := tx.packTxFields()
+	if err != nil {
+		return nil, err
+	}
+	encoded, err := EncodeRLP(packedFields)
+	if err != nil {
+		return nil, err
+	}
+	return encoded, nil
+}
+
+func (tx *LegacyEthTx) Signature() (*typescrypto.Signature, error) {
+	r := tx.R.Int.Bytes()
+	s := tx.S.Int.Bytes()
+	v := tx.V.Int.Bytes()
+
+	sig := append([]byte{}, padLeadingZeros(r, 32)...)
+	sig = append(sig, padLeadingZeros(s, 32)...)
+	if len(v) == 0 {
+		sig = append(sig, 0)
+	} else {
+		sig = append(sig, v[0])
+	}
+	if len(sig) != 65 {
+		return nil, fmt.Errorf("signature is not 65 bytes")
+	}
+
+	// pre-pend a one byte marker so nodes know that this is a legacy transaction
+	sig = append([]byte{LegacyEthTxPrefix}, sig...)
```

Comment:
@Stebalien We prepend the tx type marker byte to the signature

Reply:
Yeah, that seems like the best way forward.